app/src/main/java/com/example/composetemplate/presentation/activities/MainActivity.kt

```diff
@@ -5,13 +5,18 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import com.example.composetemplate.managers.FirebaseConfigurationManager
 import com.example.composetemplate.navigation.Navigator
 import com.example.composetemplate.ui.theme.ComposeTemplateTheme
+import org.koin.android.ext.android.inject
 
 class MainActivity : ComponentActivity() {
 
+    private val firebaseConfigurationManager: FirebaseConfigurationManager by inject()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        firebaseConfigurationManager.initConfiguration()
```

Comment:
I put this line in the MainApplication where koin is initialize instead of the MainActivity because its the first thing we need. configuration will get us strings as well and maybe we need it to display error screen before the MainActivity is called? and maybe in the future we will change to other activity to come before MainActivity and we will need to remember to move it

Reply:
OkmI will try